package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,446 @@
+        self.current_frame = frame
+
+    def _write_header(self, remarks, int n_atoms, int starting_step, 
+                      int ts_between_saves, double time_step,
+                      int charmm):
+
+        if not self.is_open:
+            raise IOError("No file open")
+
+        if not self.mode=='w':
+            raise IOError("Incorrect file mode for writing.")
+
+        #cdef char c_remarks
+        cdef int len_remarks = 0
+        cdef int with_unitcell = 1
+
+
+        if isinstance(remarks, six.string_types):
+            try:
+                remarks = bytearray(remarks, 'ascii')
+            except UnicodeDecodeError:
+                remarks = bytearray(remarks)
+
+        ok = write_dcdheader(self.fp, remarks, n_atoms, starting_step,
+                             ts_between_saves, time_step, with_unitcell,
+                             charmm)
+        if ok != 0:
+            raise IOError("Writing DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+    def write(self, xyz,  box, int step, int natoms,
+              int ts_between_saves, int charmm, double time_step, remarks):
+        """write one frame into DCD file.
+
+        Parameters
+        ----------
+        xyz : array_like, shape=(n_atoms, 3)
+            cartesion coordinates
+        box : array_like, shape=(6)
+            Box vectors for this frame
+        step : int
+            current step number, 1 indexed
+        time : float
+            current time
+        natoms : int
+            number of atoms in frame
+
+        Raises
+        ------
+        IOError
+        """
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'w':
+            raise IOError('File opened in mode: {}. Writing only allowed '
+                               'in mode "w"'.format('self.mode'))
+
+        #cdef double [:,:] unitcell = box
+        xyz = np.asarray(xyz, order='F', dtype=np.float32)
+        cdef DOUBLE_T[::1] c_box = np.asarray(box, order='C', dtype=np.float64)
+
+        if c_box.size != 6:
+            raise ValueError("box size is wrong should be 6, got: {}".format(box.size))
+
+        if xyz.shape != (natoms, 3):
+            raise ValueError("xyz shape is wrong should be (natoms, 3), got:".format(xyz.shape))
+
+        cdef FLOAT_T[::1] x = xyz[:, 0]
+        cdef FLOAT_T[::1] y = xyz[:, 1]
+        cdef FLOAT_T[::1] z = xyz[:, 2]
+        cdef float alpha, beta, gamma, a, b, c;
+
+        if self.current_frame == 0:
+            self._write_header(remarks=remarks, n_atoms=xyz.shape[0], starting_step=step,
+                               ts_between_saves=ts_between_saves,
+                               time_step=time_step,
+                               charmm=charmm)
+            self.n_atoms = xyz.shape[0]
+
+        # looks like self.nsavc is just 0 all the time
+        step = self.current_frame * self.nsavc
+
+        # we only support writing charmm format unit cell info
```

Comment:
@tylerjereddy sure this is a good idea? Doesn't NAMD write the normal unit-cell information we normally store in ts.dimensions. I think it might be nicer if we can just handle it this with giving it a standard `ts.dimenions`